test/network.spec.js

```diff
@@ -197,6 +197,17 @@ module.exports.addTests = function({testRunner, expect}) {
       expect(redirectChain.length).toBe(1);
       expect(redirectChain[0].url()).toContain('/foo.html');
     });
+    it('request.isNavigationRequest() should work', async({page, server}) => {
+      const requests = [];
+      page.on('request', request => requests.push(request));
+      server.setRedirect('/rrredirect', '/frames/one-frame.html');
+      await page.goto(server.PREFIX + '/rrredirect');
+      expect(requests[0].isNavigationRequest()).toBe(true);
+      expect(requests[1].isNavigationRequest()).toBe(true);
+      expect(requests[2].isNavigationRequest()).toBe(true);
+      expect(requests[3].isNavigationRequest()).toBe(false);
+      expect(requests[4].isNavigationRequest()).toBe(false);
```

Comment:
How do we know the order of these? Looks flaky.

Reply:
It's very deterministic:

- `requests[0]` is the initial request for the main document
- `requests[1]` is the redirected request for the main document
- `requests[2]` is the only request from the main document - the iframe main resource
- `requests[3]` and `requests[4]` are both subresources of the iframe.